### PR TITLE
[SPARKTA-255] No-time-dependant aggregations

### PR DESCRIPTION
**Pull request by danielcsant**
Now we don't depend of the time to aggregate data. 

##### How to aggregate without time
To do it you have to specify it in the policy, in the "timeDimension" field you have to put "none
  ```
   "cubes": [
    {
      "name": "testCube",
      "checkpointConfig": {
        "timeDimension": "none",
        "granularity": "minute",
        "interval": 30000,
        "timeAvailability": 60000
      }
```

##### Test 
All passed. 
Test-at updated

### CODE REVIEW

File: aggregator/src/main/scala/com/stratio/sparkta/aggregator/Cube.scala [hunk 1]
```diff
@@ -149,22 +168,37 @@ case class Cube(name: String,
...
-
-    dimensionsValues.checkpoint(new Duration(checkpointInterval))
-
-    val newUpdateFunc = (iterator: Iterator[(DimensionValuesTime, Seq[AggregationsValues], Option[Measures])]) => {
-
+  private def updateFuncAssociativeWithTime =
+    (iterator: Iterator[(DimensionValuesTime, Seq[AggregationsValues], Option[Measures])]) => {
       val eventTime =
-        DateOperations.dateFromGranularity(DateTime.now(), checkpointGranularity) - checkpointTimeAvailability
+        DateOperations
```
Comment (sgomezg) [inline aggregator/src/main/scala/com/stratio/sparkta/aggregator/Cube.scala]: Same val as https://github.com/Stratio/Sparkta/pull/1372/files#diff-57a7c91ad3fd308f93b90e866926fa4dR112

Please use only one

**Comment by sgomezg** [comment]
Very nice!!
+1

**Comment by danielcsant** [comment]
changes done. Thanks @sgomezg

File: aggregator/src/main/scala/com/stratio/sparkta/aggregator/Cube.scala [hunk 2]
```diff
@@ -149,22 +170,34 @@ case class Cube(name: String,
...
-
-    val newUpdateFunc = (iterator: Iterator[(DimensionValuesTime, Seq[AggregationsValues], Option[Measures])]) => {
+  private def updateFuncAssociativeWithTime =
+    (iterator: Iterator[(DimensionValuesTime, Seq[AggregationsValues], Option[Measures])]) => {
+      dateFromGranularity
 
-      val eventTime =
-        DateOperations.dateFromGranularity(DateTime.now(), checkpointGranularity) - checkpointTimeAvailability
+      iterator
+        .filter { case (dimensionValueTime, _, _) => dimensionValueTime.timeConfig.get.eventTime >= dateFromGranularity }
```
Comment (gserranojc) [inline aggregator/src/main/scala/com/stratio/sparkta/aggregator/Cube.scala]: I dont like this: timeConfig.get, if it is None?

File: driver/src/main/scala/com/stratio/sparkta/driver/SparktaJob.scala [hunk 3]
```diff
@@ -175,16 +175,17 @@ object SparktaJob extends SLF4JLogging {
...
 
-      Cube(name,
-        dimensions,
-        operators,
-        timeName,
-        cube.checkpointConfig.interval,
-        cube.checkpointConfig.granularity,
-        cube.checkpointConfig.timeAvailability)
+      val expiringDataConfig = timeName match {
+        case "none" => None
```
Comment (gserranojc) [inline driver/src/main/scala/com/stratio/sparkta/driver/SparktaJob.scala]: use timeName.toLowercase

File: plugins/output-elasticsearch/src/main/scala/com/stratio/sparkta/plugin/output/elasticsearch/ElasticSearchOutput.scala [hunk 4]
```diff
@@ -120,9 +120,13 @@ class ElasticSearchOutput(keyName: String,
 
   //scalastyle:on
 
-  def filterDateTypeMapping(structField: StructField, timeField: String): StructField = {
-    if (structField.name.equals(timeField)) Output.getFieldType(dateType, structField.name, structField.nullable)
-    else structField
+  def filterDateTypeMapping(structField: StructField, timeField: Option[String]): StructField = {
+    timeField match {
+      case Some(_) if structField.name.equals(timeField.get) =>
```
Comment (gserranojc) [inline plugins/output-elasticsearch/src/main/scala/com/stratio/sparkta/plugin/output/elasticsearch/ElasticSearchOutput.scala]: case Some(time) if structField.name.equals(time)

**Comment by gserranojc** [comment]
Very nice!! only 3 details, goooood job

**Pull request merged**